File: EAU.Signing/Models/BtrustPullingResult.cs
namespace EAU.Signing.Models
{
    /// <summary>
    /// Пулване на BTrust за да разберем дали документа е подписан или отказа от потребител през мобилното приложение на BTrust.
    /// </summary>
    public class BtrustPullingResult
    {
        /// <summary>
        /// Код.
        /// </summary>
        public string Code { get; set; }

        /// <summary>
        /// Статус.
        /// </summary>
        public BtrustDocStatus? Status { get; set; }

        /// <summary>
        /// Причина за отхвърляне.
        /// </summary>
        public string RejectReson { get; set; }
    }

    /// <summary>
    /// Статус на документи за BTrust: 0 = Подписан; 1 = Отказан;
    /// </summary>
    public enum BtrustDocStatus
    {
        /// <summary>
        /// Подписан.
        /// </summary>
        SIGNED = 0,

        /// <summary>
        /// Отказан.
        /// </summary>
        REJECTED = 1,
    }
}

File: EAU.Web.Portal.App/client-app/src/eau-core/models/sign/BtrustPullingResult.ts
import { TypeSystem } from 'cnsys-core';
import { moduleContext } from '../../ModuleContext';

export interface BtrustPullingResult {
    code: string;
    status?: BtrustDocStatus;
    rejectReson: string;
}

export enum BtrustDocStatus {
    SIGNED = 0,
    REJECTED = 1
}
TypeSystem.registerEnumInfo(BtrustDocStatus, 'BtrustDocStatus', moduleContext.moduleName);